client/notices/index.js

```diff
@@ -133,17 +157,19 @@ const notices = {
 	},
 
 	/**
-	 * Callback handler to clear notices when a user leaves current page
+	 * Callback middleware to clear notices when a user leaves current page
 	 * @public
+	 * @param {Object} context The page context
+	 * @param {Function} next The continuation
 	 */
 	clearNoticesOnNavigation: function( context, next ) {
 		debug( 'clearNoticesOnNavigation' );
 		let length,
 			container,
-			changed = false,
-			isNoticePersistent = function( notice ) {
-				return notice.persistent;
-			};
+			changed = false;
+		const isNoticePersistent = function( notice ) {
+			return notice.persistent;
+		};
```

Comment:
While we're fixing the ESLint warnings here, a more modern ES6 way of using the `length` and `container` variables is to restrict them to the `for` loop's scope:
```js
for ( const container in list.containerNames ) {
  const { length } = list[ container ];
}
```
This wasn't possible before ES6, because the loop didn't have its own scope.

Reply:
done